src/main/java/ctcmodel/CTCGUI.java

```diff
@@ -867,28 +879,29 @@ public void actionPerformed(ActionEvent ae) {
     
     static void fillTrackInfo(int blockID){
         //fill static info
-        StaticBlock staticBlock = trackModel.getStaticBlock(blockID);
-        trackIDText.setText("" + staticBlock.getBlockID());
-        trackSpeedText.setText("" + staticBlock.getSpeedLimit() + " mph");
+        StaticBlock staticBlock = TrackModel.getStaticBlock(blockID);
+        StaticSwitch staticSwitch = staticBlock.getStaticSwitch();
+        trackIDText.setText("" + staticBlock.getId());
+        trackSpeedText.setText("" + Convert.metersToFeet(staticBlock.getSpeedLimit())*3600.0/5280.0 + " mph");//convert units
         trackLengthText.setText("" + staticBlock.getLength() + " ft");
         trackGradeText.setText("" + staticBlock.getGrade() + "%");
         trackElevationText.setText("" + staticBlock.getElevation() + " ft");
         trackUndergroundText.setText("" + staticBlock.isUnderground());
-        boolean hasLight = staticBlock.hasLight();
-        boolean hasSwitch = staticBlock.hasSwitch();
+        //boolean hasLight = staticBlock.hasLight();
+        //boolean hasSwitch = staticBlock.hasSwitch();
+        boolean hasSwitch = (staticSwitch != null);
         boolean hasRailway = staticBlock.hasRailway();
         boolean hasHeater = staticBlock.hasHeater();
-        if(!hasLight){
-            trackLightText.setText("No Lights");
-        }
+        String stationName = staticBlock.getStation();
         if(!hasSwitch){
             trackSwitchText.setText("No Switch");
+            trackLightText.setText("No Lights");
         }
-        if(!staticBlock.hasStation()){
+        if(stationName == null){
```

Comment:
I'm not sure, but I think that blocks with no stations on them have an empty string returned instead if being null